src/terminal/parser/ut_parser/OutputEngineTest.cpp

```diff
@@ -596,6 +596,110 @@ class Microsoft::Console::VirtualTerminal::OutputEngineTest final
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:111/1/1", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1111/1/1", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/11/1", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/111/1", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/1111/1", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/1/11", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/1/111", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/1/1111", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x11, 0x11));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/23/4", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x23, 0x44));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/23/45", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x23, 0x45));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1/23/456", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x11, 0x23, 0x45));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:12/34/5", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x55));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:12/34/56", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x56));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:12/345/67", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x67));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:12/345/678", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x34, 0x67));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:123/456/789", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:123/4564/789", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:123/4564/7897", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
+        VERIFY_IS_TRUE(OutputStateMachineEngine::s_ParseColorSpec(L"rgb:1231/4564/7897", color));
+        VERIFY_ARE_EQUAL(color, RGB(0x12, 0x45, 0x78));
```

Comment:
Is it normal that the last 2 digits does not end up change the result of the RGB value? I mean `0x12` , `0x123` and `0x1231` are all `0x12` at the end. Wonder what the actual usage of `RRRRGGGGBBBB` anyway.

Reply:
that's normal. once we support >8-bit color channels (which we will probably not until 2025 or 2030), we can fix these :smile: